src/ui/public/kuery/lib/filter_migration/lib/__tests__/exists.js

```diff
@@ -0,0 +1,35 @@
+import expect from 'expect.js';
+import { convertExistsFilter } from '../exists';
+
+describe('filter to kuery migration', function () {
+
+  describe('exists filter', function () {
+
+    it('should return a kuery node equivalent to the given filter', function () {
+      const filter = {
+        meta: {
+          type: 'exists',
+          key: 'foo',
+        }
+      };
+      const result = convertExistsFilter(filter);
+
+      expect(result).to.have.property('type', 'function');
+      expect(result).to.have.property('function', 'exists');
+      expect(result.arguments[0].value).to.be('foo');
+    });
+
+    it('should return undefined if the given filter is not of type "exists"', function () {
```

Comment:
I suppose it's just one of those things that comes down to personal preference. I have the exact same feelings about exceptions XD

If only javascript had proper interfaces and polymorphism it would be a non-issue because we wouldn't have to write this hacky switching logic ourselves.